src/python/WMComponent/RucioInjector/RucioInjectorPoller.py

```diff
@@ -0,0 +1,426 @@
+        except Exception as ex:
+            msg = "Caught unexpected exception in PhEDExInjector. Details:\n%s" % str(ex)
+            logging.exception(msg)
+            raise RucioInjectorException(msg)
+
+        return
+
+    def insertContainers(self, uninjectedData):
+        """
+        This method will insert containers into Rucio, provided they cannot be found in
+        the local cache.
+        :param uninjectedData: same data as it's returned from the uninjectedFiles
+        :return: set of containers successfully inserted into Rucio
+        """
+        logging.info("Preparing to insert containers into Rucio...")
+        newContainers = set()
+        for location in uninjectedData:
+            for container in uninjectedData[location]:
+                if container not in self.containersCache:
+                    if self.rucio.createContainer(container):
+                        logging.info("Container %s inserted into Rucio", container)
+                        newContainers.add(container)
+                    else:
+                        logging.error("Failed to create container: %s", container)
+        logging.info("Successfully inserted %d containers into Rucio", newContainers)
+        return newContainers
+
+    def insertBlocks(self, uninjectedData):
+        """
+        This method will insert blocks into Rucio and attach them to their correspondent
+        containers, when attaching this block, we also need to provide the RSE that it
+        will be available.
+        :param uninjectedData: same data as it's returned from the uninjectedFiles
+        :return: a dictionary of successfully inserted blocks and their correspondent location
+        """
+        logging.info("Preparing to insert blocks into Rucio...")
+        newBlocks = {}
+        for location in uninjectedData:
+            # FIXME TODO REMOVE hack to test RSEs
+            rseName = "%s_Test" % location
+            for container in uninjectedData[location]:
+                for block in uninjectedData[location][container]:
+                    if block not in self.blocksCache:
+                        if self.rucio.createBlock(block, rse=rseName, meta=self.metaData):
+                            logging.info("Block %s inserted into Rucio", block)
+                            newBlocks[block] = location
+                        else:
+                            logging.error("Failed to create block: %s", block)
+        logging.info("Successfully inserted %d blocks into Rucio", newBlocks)
+        return newBlocks
+
+    # TODO: this will likely go away once the phedex to rucio migration is over
+    def _isBlockTierAllowed(self, blockName):
+        """
+        Check the block datatier against the list of data tiers that we want
+        to get pinned in Rucio through a block rule
+        :return: boolean whether we can or not insert a rule
+        """
+        for skipTier in self.skipRulesForTiers:
```

Comment:
You can simplify this a bunch:
endBlock = 
endTier = 
if endTier in self.skipRules: return False

Reply:
Definitely! Let me fix it.